demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,526 @@
+                inet_ntop( pServerInfo->sa_family,
+                           &( ( struct sockaddr_in6 * ) pServerInfo )->sin6_addr,
+                           resolvedIpAddr,
+                           sizeof( resolvedIpAddr ) );
+            }
+
+            LogInfo( ( "Attempting to connect to server: Host=%s, IP address=%s.",
+                       SERVER_HOST, resolvedIpAddr ) );
+
+            returnStatus = connect( *pTcpSocket, pServerInfo, serverInfoLength );
+
+            if( returnStatus == -1 )
+            {
+                LogError( ( "Failed to connect to server: Host=%s, IP address=%s.",
+                            SERVER_HOST, resolvedIpAddr ) );
+                close( *pTcpSocket );
+            }
+            else
+            {
+                LogInfo( ( "Connected to IP address: %s.",
+                           resolvedIpAddr ) );
+                break;
+            }
+        }
+
+        if( pIndex == NULL )
+        {
+            /* Fail if no connection could be established. */
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Could not connect to any resolved IP address from %.*s.\n",
+                        ( int ) strlen( pServer ),
+                        pServer ) );
+        }
+        else
+        {
+            returnStatus = EXIT_SUCCESS;
+            LogInfo( ( "Established TCP connection: Server=%.*s.\n",
+                       ( int ) strlen( pServer ),
+                       pServer ) );
+        }
+    }
+    else
+    {
+        LogError( ( "Could not resolve host %.*s.\n",
+                    ( int ) strlen( pServer ),
+                    pServer ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    /* Set the socket option for send and receive timeouts. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        transportTimeout.tv_sec = 0;
+        transportTimeout.tv_usec = ( TRANSPORT_SEND_RECV_TIMEOUT_MS * 1000 );
+
+        /* Set the receive timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_RCVTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket receive timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+
+        /* Set the send timeout. */
+        if( setsockopt( *pTcpSocket,
+                        SOL_SOCKET,
+                        SO_SNDTIMEO,
+                        ( char * ) &transportTimeout,
+                        sizeof( transportTimeout ) ) < 0 )
+        {
+            LogError( ( "Setting socket send timeout failed." ) );
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    if( pListHead != NULL )
+    {
+        freeaddrinfo( pListHead );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t transportSend( HTTPNetworkContext_t * pContext,
```

Comment:
Same comment as for the connect method and at line 329 - these should be a common library which can be used by all of your tests of TCP-based protocols.  As a bonus, app developers can use them as well.

Reply:
Yes that's the plan: Make a common library for connect and transport interface, then update these demos after the PRs are merged in as discussed with @leegeth.

Reply:
Agree and same comment applies as above. We will factor out transport implementation later. We also need to change HTTPNetworkContext_t to generic NetworkContext_t so that both MQTT and HTTP can use this. We have a separate task to handle that.